modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/entry/ViewEntryUpperToolbar.es.js

```diff
@@ -44,12 +43,13 @@ function ViewEntryUpperToolbar({
 	};
 
 	const onDelete = () => {
-		confirmDelete('/o/data-engine/v2.0/data-records/')({
+		const successMessage = Liferay.Language.get('an-entry-was-deleted');
+
+		confirmDelete('/o/data-engine/v2.0/data-records/', {successMessage})({
 			id: dataRecordId,
 		}).then((confirmed) => {
 			if (confirmed) {
-				successToast(Liferay.Language.get('an-entry-was-deleted'));
-				history.push('/');
+				setTimeout(() => history.push('/'), 500);
```

Comment:
could you please remove setTimeout?

Reply:
it smells like a workaround

Reply:
This is a workaround to prevent the refetch error after delete an entry. It was breaking on infinite loading.
Have you a suggestion for this scenario?